From f124e690fdc930de071a6419d55c6d565d0be0ac Mon Sep 17 00:00:00 2001
From: Chanh Nguyen <chanh@os.amperecomputing.com>
Date: Wed, 28 Dec 2022 16:20:12 +0700
Subject: [PATCH] timedate: Remove RTC sync down in SetTime method

If the system time is set with the SetTime method, the RTC
will be updated as well.

This commit will prevent that, the RTC will not be updated
when the SetTime method is called.

Upstream-Status: Pending

Signed-off-by: Chanh Nguyen <chanh@os.amperecomputing.com>
---
 src/timedate/timedated.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/timedate/timedated.c b/src/timedate/timedated.c
index 373574cc06..083c0e8041 100644
--- a/src/timedate/timedated.c
+++ b/src/timedate/timedated.c
@@ -895,11 +895,11 @@ static int method_set_time(sd_bus_message *m, void *userdata, sd_bus_error *erro
         }
 
         /* Sync down to RTC */
-        localtime_or_gmtime_r(&ts.tv_sec, &tm, !c->local_rtc);
+//        localtime_or_gmtime_r(&ts.tv_sec, &tm, !c->local_rtc);
 
-        r = clock_set_hwclock(&tm);
-        if (r < 0)
-                log_debug_errno(r, "Failed to update hardware clock, ignoring: %m");
+//        r = clock_set_hwclock(&tm);
+//        if (r < 0)
+//                log_debug_errno(r, "Failed to update hardware clock, ignoring: %m");
 
         log_struct(LOG_INFO,
                    "MESSAGE_ID=" SD_MESSAGE_TIME_CHANGE_STR,
-- 
2.17.1

